src/modules/job-manager/jobtap.c

```diff
@@ -1563,36 +1564,113 @@ static struct job * jobtap_lookup_active_jobid (flux_plugin_t *p,
     return job;
 }
 
+/* Job aux items are not stored in the job aux container directly, to avoid
+ * segfaults that might result from registering a destructor resident in a
+ * plugin that could be unloaded before the item is destroyed.
+ *
+ * Instead, each plugin stores one item named "jobtap::<uid>" which contains
+ * an aux container, and the actual items are stored in the inner container.
+ * When the outer container is destroyed, its destructor checks if the
+ * plugin uuid is still loaded.  If it is not, the inner aux container
+ * is leaked with a LOG_ERR message.
+ */
```

Comment:
Yes I can add that I think - that would be preferable.

While we're scanning all active jobs should we also give the plugin a `job.destroy` notification so it can manage its own state if it so chooses?

Reply:
Ah, not a bad idea though I'd have to think through the potential side effects. E.g. if a plugin is managing a job count using this method it could end up prematurely releasing some jobs if that races with the unload. I could be overthinking it, though.

Reply:
Meh I'm not sure `job.destroy`on unload is the right thing.  Maybe we should keep this focused on the problem at hand and take that up in #4392 or in some future PR.

Reply:
Yeah, I think it makes sense to announce existing jobs to plugins on load, but I'm not sure it makes sense to call `job.destroy` for jobs that are still active, even if the plugin is unloading. There may be some assumptions in the plugin about what that means for maintaining its state.

Reply:
That sounds good!

Reply:
I went ahead and squashed the fixups.

Although we could do away with the hash of plugins by uuid, it seems like it's not a bad defensive measure to check that the plugin is loaded before calling the destructor so I left the infrastructure for making that check in place.  I could be persuaded otherwise if you feel this just adds noise @grondo.

Reply:
Yeah, that seems fine to me!